mlrun/db/httpdb.py

```diff
@@ -3264,6 +3266,41 @@ def _wait_for_background_task_to_end(bg_name):
 
         return bg_task
 
+    def get_datastore_profile(
+        self, name: str, project: str
+    ) -> Optional[mlrun.common.schemas.DatastoreProfile]:
+        project = project or config.default_project
+        path = self._path_of("projects", project, "datastore_profiles") + f"/{name}"
+
+        res = self.api_call(method="GET", path=path)
+        if res and res._content:
+            public_wrapper = json.loads(res._content)
+            datastore = DatastoreProfile2Json.create_from_json(
+                public_json=public_wrapper["body"]
+            )
+            return datastore
+        return None
+
+    def delete_datastore_profile(self, name: str, project: str):
+        pass
+
+    def list_datastore_profile(
+        self, project: str
+    ) -> List[mlrun.common.schemas.DatastoreProfile]:
+        pass
+
+    def store_datastore_profile(
+        self, profile: mlrun.common.schemas.DatastoreProfile, project: str
+    ):
+        """
+        Create or replace a datastore profile.
+        :returns: None
+        """
+        project = project or config.default_project
+        path = self._path_of("projects", project, "datastore_profiles")
+
+        self.api_call(method="PUT", path=path, body=json.dumps(profile.dict()))
```

Comment:
Instead of the `body` argument, use `json=profile.dict()`. If you insist on using `json.dumps()` then please use `dict_to_json()` which defines a special encoder and is used in other places in this class.